File: 1929.concatenation-of-array.java
/*
 * @lc app=leetcode id=1929 lang=java
 *
 * [1929] Concatenation of Array
 */

// @lc code=start
class Solution {
    public int[] getConcatenation(int[] nums) {
        int[] ret = new int[nums.length * 2];
        for (int ix = 0; ix != nums.length; ++ix) {
            ret[ix] = nums[ix];
            ret[ix + nums.length] = nums[ix];
        }
        return ret;
    }
}
// @lc code=end

File: 1929.concatenation-of-array.cpp
/*
 * @lc app=leetcode id=1929 lang=cpp
 *
 * [1929] Concatenation of Array
 */

// @lc code=start
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        auto sz = nums.size();
        nums.resize(sz * 2);
        copy(nums.begin(), nums.begin() + sz, nums.begin() + sz);
        return nums;
    }
};
// @lc code=end
